lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/traffic/LambdaTrafficUpdateVerificationTask.java

```diff
@@ -64,7 +68,38 @@ public TaskResult execute(@NotNull StageExecution stage) {
             return formErrorTaskResult(stage,op.getErrors().getMessage());
         }
 
+        if (!"$WEIGHTED".equals(stage.getContext().get("deploymentStrategy"))) {
+            boolean valid = validateWeights(stage);
+            if (!valid) {
+                formErrorTaskResult(stage, "Could not update weights in time");
+                return TaskResult.builder(ExecutionStatus.TERMINAL).outputs(stage.getOutputs()).build();
+            }
+        }
+
         copyContextToOutput(stage);
         return taskComplete(stage);
     }
+
+    private boolean validateWeights(StageExecution stage) {
```

Comment:
javadoc please.  Also, since if it's possible to fail in some way other than a timeout, perhaps some use a data type for the return value that reflects that?  Maybe this doesn't matter since there isn't an ExecutionStatus.TIMEOUT as far as I can see.

Reply:
done